app/src/lib/stores/app-store.ts

```diff
@@ -889,6 +889,31 @@ export class AppStore {
     }
   }
 
+  private shouldBackgroundFetch(repository: Repository): boolean {
+    const gitStore = this.getGitStore(repository)
+    const lastFetched = gitStore.lastFetched
+
+    if (!lastFetched) {
+      return true
+    }
+
+    const now = new Date()
+    const timeSinceFetch = now.getTime() - lastFetched.getTime()
+    const twoMinutes = 2 * 60 * 1000
```

Comment:
What do think about pulling this into a top level constant, named something semantic like `BackgroundFetchMinimumInterval` or something?